functional/sev/run.sh

```diff
@@ -0,0 +1,352 @@
+  esudo tee Dockerfile <<EOF
+FROM alpine:3.14
+RUN apk update && apk upgrade && apk add openssh-server
+RUN ssh-keygen -t ed25519 -f /etc/ssh/ssh_host_ed25519_key -P ""
+# A password needs to be set for login to work. An empty password is
+# unproblematic as password-based login to root is not allowed.
+RUN passwd -d root
+# Generate with \`ssh-keygen -t ed25519 -f ccv0-ssh -P "" -C ""\`
+COPY ccv0-ssh.pub /root/.ssh/authorized_keys
+ENTRYPOINT /usr/sbin/sshd -D
+EOF
+
+  # Build image
+  esudo docker build -t ccv0-ssh .
+
+  # Upload image
+  esudo docker tag ccv0-ssh:latest ${DOCKER_USER}/ccv0-ssh:unencrypted
+  esudo docker push ${DOCKER_USER}/ccv0-ssh:unencrypted
+
+  popd
+}
+
+encrypt_image() {
+  # Create key file
+  esudo tee test/keys.json <<EOF
+{
+  "key_id1":"RcHGava52DPvj1uoIk/NVDYlwxi0A6yyIZ8ilhEX3X4="
+}
+EOF
+
+  # Run skopeo command to encrypt new image - ubuntu-rootfs-osbuilder-sev-encrypted:encrypted
+  esudo OCICRYPT_KEYPROVIDER_CONFIG=attestation-agent/sample_keyprovider/src/enc_mods/offline_fs_kbs/ocicrypt.conf \
+    skopeo copy docker-daemon:${DOCKER_USER}/ccv0-ssh:unencrypted \
+    docker-daemon:${DOCKER_USER}/ccv0-ssh:encrypted \
+    --encryption-key provider:attestation-agent:test/keys.json:key_id1
+
+  # Push encrypted image
+  esudo docker push ${DOCKER_USER}/ccv0-ssh:encrypted
+}
+
+calculate_measurement_and_add_to_kbs() {
+  # Generate digest from sev-snp-measure output - this also inserts measurement values inside OVMF image
+  mout=$(esudo sev-snp-measure -v --mode=sev --output-format=base64 \
+    --ovmf="${EDK2_REPO_DIR}/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd" \
+    --kernel="/usr/share/kata-containers/vmlinuz-sev.container" \
+    --initrd="/usr/share/kata-containers/kata-containers-sev-initrd.img" \
+    --append="tsc=reliable no_timer_check rcupdate.rcu_expedited=1 i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k cryptomgr.notests net.ifnames=0 pci=lastbus=0 console=hvc0 console=hvc1 debug panic=1 nr_cpus=1 scsi_mod.scan=none agent.log=debug agent.log=debug initcall_debug agent.config_file=/etc/default/agent.cfg console=ttyAMA0,115200 console=tty highres=off console=ttyS0" \
+  )
+
+  # Trim text and error if value is empty
+  measurement=$(echo ${mout} | sed "s/^\(Calculated SEV guest measurement: \)\(.*\)$/\2/g")
+  if [[ -z "${measurement}" ]]; then >&2 echo "Measurement is invalid"; return 1; fi
+
+  # Add key, keyset and policy with measurement to DB
+  mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} -e "INSERT INTO secrets VALUES (10, 'key_id1', 'RcHGava52DPvj1uoIk/NVDYlwxi0A6yyIZ8ilhEX3X4=', 10);"
+  mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} -e "INSERT INTO keysets VALUES (10, 'KEYSET-1', '[\"key_id1\"]', 10);"
+  mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} -e "INSERT INTO policy VALUES (10, '[\"${measurement}\"]', '[]', 0, 0, '[]', now(), NULL, 1);"
```

Comment:
I think you can do something like this:

```
mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} <<EOF
  INSERT INTO secrets VALUES (10, 'key_id1', 'RcHGava52DPvj1uoIk/NVDYlwxi0A6yyIZ8ilhEX3X4=', 10);
  INSERT INTO keysets VALUES (10, 'KEYSET-1', '["key_id1"]', 10);
  INSERT INTO policy VALUES (10, '["${measurement}"]', '[]', 0, 0, '[]', now(), NULL, 1);
EOF
```

It might be a bit more readable, and no need to escape `"` chars.

Reply:
done